### PR TITLE
Finally get rid of the dependencies and cleanup

### PR DESCRIPTION
**Pull request by costinm**
- consistent checkout: checkout, attach workspace ( for binaries can cached stuff from build ), make sync
(make sync combines the previous submodules steps and init)
 
- depend on test and build ( nightly was failing and probably slower than needed)

- remove the debug statements for attach workspace

- remove dependencies target, thanks to vendor no longer needed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -590,41 +565,52 @@ workflows:
...
             - test
+            - build
       # Nightly release
       - nightly:
           context: org-global
           requires:
             - e2e-simple
             - e2e-pilot
-            - e2e-pilot-auth
+            - e2e-pilot-alpha3
```
Comment (ldemailly) [inline .circleci/config.yml]: keep both ?

Comment (costinm) [reply .circleci/config.yml]: You know me: yes, we will have both, but right now pilot-auth is failing ( with a legitimate bug), once the bug is fixed I'll add it back. I don't think we should stop building nightly while auth is failing or flaky.

Comment (ldemailly) [reply .circleci/config.yml]: well we disagree - what would a nightly be useful for if it doesn't show the state of every major feature, which auth is a key one.

Comment (costinm) [reply .circleci/config.yml]: It shows the real state of the master - good or broken. 
It's a nightly, not a release - sometimes some features are broken. 

In this case - auth has one bug in a particular corner case - not a good reason to 
stop building nightlies ( which can be used to find other regressions and bugs ).

We also have dailies - which are only build if everything is perfect. No point of making 
the nighty identical with the daily, they serve different purpose.

Comment (costinm) [reply .circleci/config.yml]: In other words: this is how CircleCI 'nightly' was intended to work. Produce a build every night,
even if some features are broken in master. 

You already have the daily build that satisfy your requirement and is designed they way you want.

Comment (ldemailly) [reply .circleci/config.yml]: by that principle why is it running any test then ? is it meant to be tweaked every day to an ever changing (or shrinking) set of tests that happens to pass that day ?

Comment (costinm) [reply .circleci/config.yml]: Not every day - but yes, it is meant to be tweaked, and the tests are run to know the status.
The goal is to reflect reality. 

I think auth is now fixed so I'll add it back after it stays green for few cycles.

Comment (ldemailly) [reply .circleci/config.yml]: this doesn't make sense/isn't scalable - we should run all the tests and report which ones failed in the test matrix

**Pull request merged**